diff --git a/Source/WTF/wtf/dtoa/utils.h b/Source/WTF/wtf/dtoa/utils.h
index 6a875fdf9..d697f6bb5 100644
--- a/Source/WTF/wtf/dtoa/utils.h
+++ b/Source/WTF/wtf/dtoa/utils.h
@@ -86,7 +86,7 @@ int main(int argc, char** argv) {
     defined(__powerpc__) || defined(__ppc__) || defined(__ppc64__) || \
     defined(_POWER) || defined(_ARCH_PPC) || defined(_ARCH_PPC64) || \
     defined(__sparc__) || defined(__sparc) || defined(__s390__) || \
-    defined(__SH4__) || defined(__alpha__) || \
+    defined(__SH4__) || defined(__alpha__) || defined(__sw_64__) || \
     defined(_MIPS_ARCH_MIPS32R2) || \
     defined(__AARCH64EL__) || defined(__aarch64__) || defined(__AARCH64EB__) || \
     defined(__riscv) || \
diff --git a/Source/WebKit/UIProcess/Launcher/glib/Syscalls.h b/Source/WebKit/UIProcess/Launcher/glib/Syscalls.h
index 18dea9a94..1fd556b83 100644
--- a/Source/WebKit/UIProcess/Launcher/glib/Syscalls.h
+++ b/Source/WebKit/UIProcess/Launcher/glib/Syscalls.h
@@ -49,6 +49,10 @@
 # define FLATPAK_MISSING_SYSCALL_BASE 0x40000000
 #endif
 
+#if defined(__sw_64__)
+# define FLATPAK_MISSING_SYSCALL_BASE (-153)
+#endif
+
 // FLATPAK_MISSING_SYSCALL_BASE:
 //
 // Number to add to the syscall numbers of recently-added syscalls
@@ -148,6 +152,8 @@
 # define __SNR_process_madvise __NR_process_madvise
 #endif
 
+#ifndef __sw_64__
+
 #ifndef __NR_epoll_pwait2
 # define __NR_epoll_pwait2 (FLATPAK_MISSING_SYSCALL_BASE + 441)
 #endif
@@ -197,4 +203,6 @@
 # define __SNR_memfd_secret __NR_memfd_secret
 #endif
 
+#endif
+
 // Last updated: Linux 5.14, syscall numbers < 448
